### ISSUE TITLE
Activity center: Accepting / declining contact requests is broken

### ISSUE THREAD

**Issue by MishkaRogachev** [status-im/status-desktop#9505]
# Bug Report

Recent changes with new contact requests logic probably caused an issues: AC notification for contact request fires just one time for a contact. If user declines first CR he won't receive notification for new CR from that user.

## Steps to reproduce

1. User *Thu9Test1* sends CR request to user *ContactRequest2*:
 - *Thu9Test1* does not receive any feedback and can send CR again
 -  *ContactRequest2* gets the AC notification with CTA  and can also respond via profile menu
 
<img width="610" alt="Screenshot 2023-02-09 at 16 38 56" src="https://user-images.githubusercontent.com/2522130/217817064-02253bc9-38d3-4273-8a84-202b814b7c70.png">
<img width="790" alt="Screenshot 2023-02-09 at 16 39 50" src="https://user-images.githubusercontent.com/2522130/217817203-c411c60e-6d93-4606-9e8a-1414643355c4.png">

2. User *ContactRequest2* declines the contact request
- *Thu9Test1* does not receive any feedback and can send CR again
- *ContactRequest2* gets the AC notification update with "Declined" state
<img width="574" alt="Screenshot 2023-02-09 at 16 40 35" src="https://user-images.githubusercontent.com/2522130/217817826-14ce88f5-0461-4663-ae4c-a4f551266192.png">

3.  User *Thu9Test1* sends CR request again
- *Thu9Test1* does not receive any feedback and can send CR again
- *ContactRequest2* gets notification sound and profile dialog updates with "Respond to CR" control. AC notification does **not** get updated. There is still old notification with declined state.

4.  User *ContactRequest2* accepts the contact request 
-  *Thu9Test1* receives the feedback notification with accepted state and *ContactRequest2* appears in contact list. 
<img width="583" alt="Screenshot 2023-02-09 at 16 40 45" src="https://user-images.githubusercontent.com/2522130/217819043-120df346-48aa-4a57-bc9f-4d59248189f4.png">

-  *ContactRequest2* gets *Thu9Test1* added to contact list but no notification update. Notification is still in the declined state

**Comment by John-44** [comment]
@MishkaRogachev Thank you for the very well written issue :-)  To clarify what is supposed to happen during steps in this flow:

At step 1, after sending the initial contact request to *ContactRequest2*, *Thu9Test1* should be unable to send a 2nd contact request via the UI (because *Thu9Test1*'s contact request to *ContactRequest2* would be in the pending state, *Thu9Test1* should only be able to update the contact request message, not send a new contact request, see the designs below):

![2023-02-10 12_22_46-Kuba⎜Desktop – Figma](https://user-images.githubusercontent.com/22846356/218091799-7039740a-81d7-4cf9-92fd-8653a2e1b1a6.png)

---

Step 2 looks all good

---

At step 3, *Thu9Test1* should be unable to send a 2nd contact request via the UI.   This is because from *Thu9Test1*'s point of view, their contact request to *ContactRequest2* would still be in the pending state (and therefor *Thu9Test1* should only be able to update the message in their initial contact request).  See design posted below my comment re. step 1, this also applies here.

Also *ContactRequest2* should not receive any notification when a contact request message is updated, because if they were to receive a notification when this happens it would become a spam vector e.g. if we were to issue notifications when contact request messages are changed, in this scenario *Thu9Test1* could keep changing their contact request message in order to spam *ContactRequest2* with notifications.

---

Step 4 as it is currently written doesn't make sense given the above.  Step 4 should be replaced with the following:

*ContactRequest2* navigates to Settings/Messaging/'Contacts, Requests, and Blocked Users/ and then clicks on the "Rejected requests" tab.  From this view, *ContactRequest2* would then click on the icon to the right of *Thu9Test1*'s name to change the previous rejection of  *Thu9Test1*'s contact request to an acceptance of  *Thu9Test1*'s contact request.  (note this icon is currently represented in the "Desktop Settings" Figma design by a red circle with a cross across it - @benjthayer @xAlisher @mariocnovoa we should probably change this icon)

![2023-02-10 12_49_24-⚙️ Settings _ Desktop – Figma](https://user-images.githubusercontent.com/22846356/218097115-f3e0b88c-8bb6-4ad9-a666-e8663d5e2f13.png)

**Comment by John-44** [comment]
@ilmotta see my comment above, the behaviour I've described in this comment should be the same for mobile

**Comment by ilmotta** [comment]
Thanks @John-44 and @MishkaRogachev for the very detailed comments.

I hope soon we'll be able to implement in Mobile the feature to accept back rejected CRs. For the Feb RC, the user won't be able to manage rejected CRs, as there's no screen for that yet.

**Comment by anastasiyaig** [comment]
@jrainville please check this out too when u have time

**Comment by John-44** [comment]
@ilmotta @MishkaRogachev @anastasiyaig @jrainville @cammellos  - I just spotted that I made an error in the comment I made above ( https://github.com/status-im/status-desktop/issues/9505#issuecomment-1425769497 ).  I have now edited that comment to correct that error.

The error I made was that I incorrectly stated that a user can edit a mutual contact request message after it has been sent.  This is wrong, a user should not be able to edit a mutual contact request message after it has been sent.   My apologies for this mistake in my previous comment.   

The previous comment is now correct (after the edit I have just made).
